main.c

```diff
@@ -9,128 +10,413 @@
+                            for (int i = 0; i < tag_count; i++) {
+                                printf("1. biography\t2. fantasy\t3. fiction\t4. mystery\n"
+                                       "5. romance\t6. scifi\t7. thriller\t8. young_adult\n");
+                                printf("Tag #%i: ", i + 1);
+                                scanf("%i", &tags[i]);
+                            }
+                        }
+                        printf("Filtered by Total Count? ");
+                        scanf("%i", &selection);
+                        if (selection == 1) {
+                            printf("Input count: ");
+                            scanf("%i", &total_count);
+                        }
+                        printf("Filtered by In-Stock Count? ");
+                        scanf("%i", &selection);
+                        if (selection == 1) {
+                            printf("Input count: ");
+                            scanf("%i", &in_stock_count);
+                        }
+                        printf("Filtered by Likes? ");
+                        scanf("%i", &selection);
+                        if (selection == 1) {
+                            printf("Input likes: ");
+                            scanf("%i", &likes);
+                        }
+                        Book *filterBy = createBook(ISBN, title, author_id, status, tags, tag_count, total_count, in_stock_count, likes);
+                        Book *filtered_books = filterBooks(filterBy, library);
+                        if (filtered_books)
+                        {
+                            printf("%i matching book(s) found!\n-----------\n", countBooks(filtered_books));
+                            printBooks(filtered_books);
+                        }
+                        else    printf("No matching book found\n");
+                        freeBookList(filtered_books);
+                        break;
+                    case 4:
+                        printf("Enter ISBN of book to be deleted: ");
+                        char* ISBN_to_delete = malloc(20);
+                        scanf("%s", ISBN_to_delete);
+                        // Delete source book
+                        int deleted_book = deleteBook(ISBN_to_delete);
+                        if (deleted_book > 0)  printf("%i book deleted\n", deleted_book);
+                        else    printf("No book found!\n");
+                        // Delete all copies of source book
+                        int deleted_book_copies = deleteBookCopies(ISBN_to_delete);
+                        if (deleted_book_copies > 0)    printf("%i book copies deleted\n", deleted_book_copies);
+                        else    printf("No book copies found!\n");
+                        free(ISBN_to_delete);
+                        break;
+                    default:
+                        break;
+                }
```

Comment:
I think we need to remove/inactive the (current) transactions of deleted book as well